Resolve issue #1090

abdala-elgendy opened the pull request:
Moved index column generation logic from create() into a dedicated helper method.
Helper Method (generateRandomComment()):
This method creates a random alphanumeric string. This string is appended after the prefix "idx_" to form a more diverse comment.

mrigger commented:
Thanks for the PR! It seems the PR only extracts code into a helper method? Or did I overlook anything?

abdala-elgendy commented:
the first time I added a method to generate random comments to diversify generated queries but the tests did not pass I thought because of this modification, when I deleted and reran the tests it didn't also pass " I'm just extracting a helper method to clarify the code in the second time"

abdala-elgendy commented:
I add MySQLBooleanConstant for boolean support.
the project ( in MySQL part ) likely did not have a dedicated way to represent TRUE and FALSE as separate boolean constants. 
MySQL treats BOOLEAN as a synonym for TINYINT(1), meaning TRUE was likely represented as 1 and FALSE as 0 in SQLancer.

![image](https://github.com/user-attachments/assets/76424dbc-e497-4bae-8b5a-ab324bfe71ea)

generated queries might have like "WHERE is_active = 1" instead of "WHERE is_active = TRUE" 

I don't yet edit the generated queries part.

mrigger commented on src/sqlancer/mysql/ast/MySQLBooleanConstant.java:
```diff
@@ -0,0 +1,77 @@
+        if (rightVal.isBoolean()) {
+            return MySQLConstant.createBoolean(this.value == rightVal.asBooleanNotNull());
+        }
+        throw new UnsupportedOperationException("Cannot compare boolean with non-boolean values.");
```
I think we should throw an `IgnoreMeException` here. This logic is primarily used by Pivoted Query Synthesis (PQS). In general, given that PQS is not widely used, I think we can simplify the implementation here and directly throw an `IgnoreMeException` rather than providing the `isBoolean` method and implementing `isEquals`. The current logic also does not seem complete or correct. For example, `SELECT 1 = TRUE` can be evaluated and evaluates to true, which the method would not cover.

mrigger commented on src/sqlancer/mysql/ast/MySQLCastOperation.java:
```diff
@@ -6,7 +6,7 @@ public class MySQLCastOperation implements MySQLExpression {
 
     public enum CastType {
-        SIGNED, UNSIGNED;
+        SIGNED, UNSIGNED , INT, BOOLEAN ,TEXT;
```
Please keep the PR focused on a single change. We can review and merge changes more quickly this way.

mrigger commented on src/sqlancer/mysql/gen/datadef/MySQLIndexGenerator.java:
```diff
@@ -160,4 +124,46 @@ private void indexType() {
     public void setNewSchema(MySQLSchema schema) {
         this.schema = schema;
     }
+    private String generateIndexColumns(MySQLTable table, MySQLExpressionGenerator gen) {
```
For such refactorings, it would be great to add a separate commit or PR. This makes reviewing easier, as it is clearer whether something is a new feature or should preserve the existing functionality.

mrigger reviewed (commented):
Thanks for the PR! I added some comments/feedback to help improve it.

abdala-elgendy commented:
I forget to delete 
![image](https://github.com/user-attachments/assets/6b09c1f4-659b-46ce-af5e-c2a95eec09bf)
so tests faild
